libacb/util/deps.go

```diff
@@ -0,0 +1,354 @@
+// Copyright 2015 The appc Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"crypto/md5"
+	"crypto/sha1"
+	"crypto/sha256"
+	"crypto/sha512"
+	"crypto/tls"
+	"fmt"
+	"hash"
+	"io"
+	"net/http"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/appc/spec/discovery"
+	"github.com/appc/spec/schema/types"
+	"github.com/coreos/ioprogress"
+)
+
+var (
+	defaultArch = "amd64"
+	defaultOS   = "linux"
+)
+
+// RenderACI will download any missing dependencies into the depstore,
+// expand any unexpanded dependencies into the scratchpath, and return
+// an ordered list of all dependencies for overlaying.
+func RenderACI(acipath, scratchpath, depstore string, insecure bool) ([]string, error) {
```

Comment:
This is because I was suggesting to use appc/spec acirenderer. Overlaying dependencies is not enough since aci may have a pathWhiteList. I'll be happy to do/help you moving this to acirenderer. Let me know if you prefer to do this in this PR or in a follow up PR.